parachain/README.md

```diff
@@ -104,13 +120,50 @@ cargo install subxt-cli
 ```
 
 Update metadata by fetching it from parachain node (in this case a node in the E2E stack):
-```
+```bash
 subxt metadata --url http://localhost:8081 -f bytes > tools/query-events/metadata.scale
 ```
 
 If you want to update the tool for an already running E2E stack:
 
-```
+```bash
 cargo build --release --manifest-path tools/query-events/Cargo.toml
 cp target/release/snowbridge-query-events /tmp/snowbridge/bin/
 ```
+
+## Generating pallet weights from benchmarks
+
+Build the parachain with the runtime benchmark flags and generate the chain spec:
+
+```bash
+runtime=snowbase
+cargo build \
+    --release \
+    --no-default-features \
+    --features "${runtime}-native,rococo-native,runtime-benchmarks,${runtime}-runtime-benchmarks" \
+    --bin snowbridge
+./target/release/snowbridge build-spec --chain $runtime --disable-default-bootnode > /tmp/snowbridge/spec.json
+cat "/tmp/snowbridge/spec.json" | node ../test/scripts/helpers/mutateSpec.js "/tmp/snowbridge/contracts.json" "/tmp/snowbridge/initialBeaconSync.json" | sponge "/tmp/snowbridge/spec.json"
```

Comment:
@claravanstaden, just checking, surely benchmarks can be run using the default builtin chainspec? ie.
```
/target/release/snowbridge build-spec --chain $runtime --disable-default-bootnode
```

Reply:
I ran the benchmarks with the chainspec generated from the `start-services.sh` script. I must confess I didn't think of using the built-in chainspec.

Reply:
ahh, ok. So I tried to run the benchmarks using the default spec, and they failed with:

```
vincent@lemur:~/code/snowbridge/parachain (main)$ target/release/snowbridge benchmark pallet --chain snowbase --execution wasm --wasm-execution compiled --pallet snowbridge_ethereum_beacoclient --extrinsic '*' --r
epeat 20 --steps 50 --template templates/module-weight-template.hbs
2022-11-08 11:27:06 💫 Sync committee size is: 32
2022-11-08 11:27:06 💫 Received initial sync, starting processing.
2022-11-08 11:27:06 Merkle proof branch length doesn't match depth.
2022-11-08 11:27:06 Initial sync failed with error Module(ModuleError { index: 18, error: [7, 0, 0, 0], message: Some("InvalidSyncCommitteeMerkleProof") })
```

The problem is that when the benchmarks tool is started with the default spec, it generates the genesis config using the default spec, and that causes this line to error out: https://github.com/Snowfork/snowbridge/blob/81348245bfeaa66b9e02a09c09aa696167bfa1d1/parachain/pallets/ethereum-beacon-client/src/lib.rs#L176. Because the default spec contains an empty `InitialSync`

I think the fix is to bypass processing of the InitialSync if its the default value. This would allow benchmarks to run normally using the default spec which is highly desirable.

Reply:
Added a ticket here: https://linear.app/snowfork/issue/SNO-369

Reply:
This is generating that same chainspec ☝️ Just without starting an E2E env. The benchmark fails if we don't run mutateSpec on the spec first: `Error: Input("Error parsing spec file: invalid type: sequence, expected a hex string at line 122 column 32")`. Looks like `.genesis.runtime.ethereumBeaconClient.initialSync.current_sync_committee.aggregate_pubkey` is an array of zeroes where the benchmark subcommand expects a hash string like `0x8fe11476a05750c52618deb79918e2e674f56dfbf12dbce55ae4386d108e8a1e83c6326f5957e2ef19137582ce270dc6`